packages/docs-site/src/content/docs/docs/guides/multi-env.mdx

```diff
@@ -3,79 +3,82 @@ title: Multi-environment configuration
 
-You may have a different env var that you want to use to switch between values. For example, you might want to use `APP_ENV` instead of `NODE_ENV`. You can do this by passing the name of the env var you want to use as the first argument to `switchBy`.
+Even in those cases, we recommend creating your own more specific env flag, for example `APP_ENV`, that you have total control over. Here is an example of building our own env flag based on the env vars injected by Vercel. In this example, we'd like to differentiate between PR previews, branch previews, and a special staging branch.
 
-```typescript
-import { switchBy } from 'dmno';
+```ts title='.dmno/config.mts'
+import { defineDmnoService, switchBy, pickFromSchemaObject } from 'dmno';
+import { VercelEnvSchema } from '@dmno/vercel-platform';
 
 export default defineDmnoService({
-  pick: [
-    'APP_ENV',
-  ],
   schema: {
-    MY_CONFIG_ITEM: {
-      value: switchBy('APP_ENV', {
-        _default: 'default value',
-        development: 'development value',
-        staging: 'staging value',
-        production: 'production value',
-      }),
+    ...pickFromSchemaObject(VercelEnvSchema, 'VERCEL_ENV', 'VERCEL_GIT_COMMIT_REF'),
+    APP_ENV: {
+      value: () => {
+        if (DMNO_CONFIG.VERCEL_ENV === 'production') return 'production';
+        if (DMNO_CONFIG.VERCEL_ENV === 'preview') {
+          if (DMNO_CONFIG.VERCEL_GIT_PULL_REQUEST_ID) return 'pr-preview';
+          else if (DMNO_CONFIG.VERCEL_GIT_COMMIT_REF === 'staging') return 'staging';
+          else return 'branch-preview';
+        }
+        return 'development';
+      },
     },
+    SOME_VAR: switchBy('APP_ENV', { /* ... */ }),
   },
 });
 ```
 
-:::note
-See the [switchBy](/docs/reference/helper-methods/#switchby) reference for more information.
-:::
+In cases where the platform may not be injecting env vars, but you need to toggle the environment, you can pass it in as an env var from the command line. For example:
+
+```bash
+APP_ENV=production pnpm exec dmno run -- your-build-command
+```
+
+## Overrides from `process.env`
+
+Keep in mind that DMNO loads in process environment variables in as _overrides_, so there are many cases where your schema does not have a value defined for a certain environment flag, because you are expecting to receive it as an override. Sometimes this could be contextual information injected by the hosting platform, like `VERCEL_GIT_PULL_REQUEST_ID`, or it could be env vars that have been set via the platform's env var management UI. This will be especially true if you are migrating an existing project to DMNO.
```

Comment:
when you do the new override guides stuff in the future, it's probably worth linking here